utils/http/http.go

```diff
@@ -37,7 +38,7 @@ func GenBody(t interface{}) io.Reader {
 
 // http post
 func PostRequest(url string, body io.Reader) ([]byte, error) {
-	client := &http.Client{}
+	client := &http.Client{Timeout: config.GetHttpTimeout()}
```

Comment:
Actually when you are using http.Client without passing http.Transport, it will be using the default http.DefaultTransport, see https://medium.com/@nate510/don-t-use-go-s-default-http-client-4804cb19f779 the timeout on client is actually coarse grained, better control need to go through http.Transport

Reply:
Good catch. It's a good idea to create an instance of `http.Transport` and use it instead of relying on the default transport. The default transport has a few surprising properties that may come back and bite you.